### PR TITLE
many: assorted shellcheck fixes

### PR DESCRIPTION
**Pull request by bboozzoo**
run-checks --static will run shellcheck only if one is installed in the system,
which is not the case when it's called in spread tests. As a result quite a lot
of issues got their way into the code unnoticed.

### CODE REVIEW

File: get-deps.sh [hunk 1]
```diff
@@ -3,18 +3,20 @@
 set -e
 
 if [ "$GOPATH" = "" ]; then
-    export GOPATH=$(mktemp -d)
+    GOPATH=$(mktemp -d)
```
Comment (mvo5) [inline get-deps.sh]: Should we have quotes around this? I know its not (strictly) needed for $() but I wonder if we should still do it for consistently. OTOH it looks like we are a bit all over the place with $().

File: get-deps.sh [hunk 2]
```diff
@@ -3,18 +3,20 @@
...
+    export GOPATH
+    # shellcheck disable=SC2064
```
Comment (zyga) [inline get-deps.sh]: This warning is https://github.com/koalaman/shellcheck/wiki/SC2064

What happens if we expand GOPATH later?

File: tests/lib/prepare.sh [hunk 3]
```diff
@@ -379,11 +383,13 @@ EOF
...
     if is_core18_system; then
-        extra_snap="$IMAGE_HOME"/snapd_*.snap
+        extra_snap=("$IMAGE_HOME"/snapd_*.snap)
     else
-        extra_snap="$IMAGE_HOME"/core_*.snap
+        extra_snap=("$IMAGE_HOME"/core_*.snap)
     fi
 
+    # shellcheck disable=SC2128
+    # extra_snap should contain only ONE snap
```
Comment (zyga) [inline tests/lib/prepare.sh]: This is https://github.com/koalaman/shellcheck/wiki/SC2128

Could we expand the first element ourselves?

File: update-pot [hunk 4]
```diff
@@ -34,7 +34,8 @@ go install github.com/snapcore/snapd/i18n/xgettext-go
 
 # exclude vendor and _build subdir
 I18N_FILES="$(mktemp -d)/i18n.files"
-find "$HERE" -type d \( -name "vendor" -o -name "_build" \) -prune -o -name "*.go" -type f -print > $I18N_FILES
+find "$HERE" -type d \( -name "vendor" -o -name "_build" \) -prune -o -name "*.go" -type f -print > "$I18N_FILES"
+# shellcheck disable=SC2064
 trap "rm -rf $(dirname "$I18N_FILES")" EXIT
```
Comment (zyga) [inline update-pot]: Same question as above, should this expand now or later?

File: tests/lib/prepare-restore.sh [hunk 5]
```diff
@@ -117,7 +118,7 @@ build_rpm() {
     cp -ra -- * "/tmp/pkg/snapd-$version/"
     mkdir -p "$rpm_dir/SOURCES"
     # shellcheck disable=SC2086
-    (cd /tmp/pkg && tar "-c${archive_compression}f" "$rpm_dir/SOURCES/$archive_name" $extra_tar_args "snapd-$version")
+    (cd /tmp/pkg && tar "-c${archive_compression}f" "$rpm_dir/SOURCES/$archive_name" "${extra_tar_args[@]}" "snapd-$version")
```
Comment (zyga) [inline tests/lib/prepare-restore.sh]: I believe this fails in practice:
```
+ tar -cJf /usr/src/packages/SOURCES/snapd_1337.2.34~pre1.vendor.tar.xz '' snapd-1337.2.34~pre1
tar: Substituting `.' for empty member name
tar: : Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
```

File: get-deps.sh (hunk 2 above)
Comment (bboozzoo) [reply get-deps.sh]: Nothing exciting, we don't modify it elsewhere, so it'll effectively have the same value as it has here.

File: tests/lib/prepare-restore.sh (hunk 5 above)
Comment (bboozzoo) [reply tests/lib/prepare-restore.sh]: Heh, shell quoting and preserving single quotes is crazy.

File: tests/lib/prepare.sh (hunk 3 above)
Comment (bboozzoo) [reply tests/lib/prepare.sh]: I need to doble check with bash. IIRC in zsh, if `var` ended up being a single string, expanding `${var[0]}` would yield the first letter of that string.

Comment (bboozzoo) [reply tests/lib/prepare.sh]: BTW. ${var} if var is an array gives the first element.

Comment (zyga) [reply tests/lib/prepare.sh]: Let's do that. I'm mainly after not having shellcheck exceptions. If you choose to do this please add a comment, I'm not sure if the particular array semantics is widely known.

**Pull request merged**